src/main/java/edu/umich/its/cpm/Utils.java

```diff
@@ -443,19 +443,46 @@ public static String getUserLoginId(HttpServletRequest request, Environment env)
 		boolean isTestUrlEnabled = Boolean.valueOf(env.getProperty(ALLOW_USER_URLOVERRIDE));
 		String testUserInSession = (String) request.getSession().getAttribute(TEST_USER);
 
-		if (isTestUrlEnabled && testUser != null) {
+		//become user case
```

Comment:
Handle the simpler case of isTestUserEnabled == false up front.  Then there isn't any reason to keep checking the value of isTestUserEnabled. 

Complicated boolean IF checks require careful analysis of what happens with each possible failure mode.